### ISSUE TITLE
Introduce Local isn't triggered when the selected expression is followed by an indexer

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#28941]
**Version Used**: 
15.8.0 Preview 5.0

**Steps to Reproduce**:
```csharp
byte[] getArray() => null;
void test()
{
    var foo = getArray()[0]; // select here
    var bar = getArray();
}
```
In the `foo`-assignment-line, select the `getArray()` sub-expression and activate the code fixes.

**Expected Behavior**:
`Introduce Local` should be offered

**Actual Behavior**:
`Introduce Local` isn't offered

**Notes**
When I select the `getArray()` expression in the `bar`-line and activate "for all occurences", then the call in the `bar`-line is recognized correctly and extracted successfully.

![introduce-local-indexer](https://user-images.githubusercontent.com/1947968/43411116-8ce821e4-9429-11e8-94d8-ef7e174e86ee.gif)
